[KristopherPTaslim] iP

KristopherPTaslim opened the pull request:
## DukePro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon

If you java programmer, you can use it to practice Java too. Here's the `main` method:

``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

KristopherPTaslim committed:
Level-1

KristopherPTaslim committed:
Finish Level-2

KristopherPTaslim committed:
Finish Level-3

KristopherPTaslim committed:
Finish Level-4

KristopherPTaslim committed:
Finish Level-5

KristopherPTaslim committed:
Finish Level-6

KristopherPTaslim committed:
Finish Testing

KristopherPTaslim committed:
Finish Branch-Level-7

KristopherPTaslim committed:
Finish branch-Level-8

KristopherPTaslim committed:
Merge branch 'branch-Level-8'

KristopherPTaslim committed:
Debug Branch Level-7

pnutzz-0207 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
```
Following the coding standards, you might need to arrange the import statements into alphabetical order! :) 
Same for some of the other classes that includes import statements.
```suggestion
import java.io.IOException;
import java.time.format.DateTimeFormatter;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Scanner;
```

pnutzz-0207 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+
+class Duke {
+
+    public static void main(String[] args) throws DukeException{
```
You might need to leave a space after _DukeException_ 
```suggestion
    public static void main(String[] args) throws DukeException {
```

pnutzz-0207 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+
+    public static void main(String[] args) throws DukeException{
+        Scanner sc = new Scanner(System.in);
+        FileClass fc = new FileClass(); //file class 
+        String home = System.getProperty("user.home"); //home directory
```
I like how you make comments behind the lines! It makes the logic clear 😄

pnutzz-0207 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                        String dateTime = input.substring(indexOfTime + deadlineCondition.length(), input.length()); // the date and time for by
+                        //convert to the correct one
+                        LocalDateTime deadlineTime = LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("d/M/y Hmm"));
+                        String convertedTime = deadlineTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a"));
```
I like how you name your variables :), it is very clear what they all do 👍

pnutzz-0207 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+
+        // Recording everything in the duke.txt
+        for (int i = 0; i < taskArray.size(); i++ ) {
+            Task tasks = taskArray.get(i);
```
Since it is referring to just 1 task, will it be better to name this variable as 'task' instead of 'tasks'?

pnutzz-0207 reviewed (commented):
Overall you are following the coding standards, very good job! 💯  just a few minor mistakes made, after you change it, it'll all look good to me 👏

manu2002g commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+    
+                    String eventCondition = "/at "; 
+                    try {
+                        int indexOfTime = input.indexOf(eventCondition); //to find /
+                        String dateTime = input.substring(indexOfTime + eventCondition.length(), input.length()); // the date and time for by
+                        //convert to the correct one
+                        LocalDateTime eventTime = LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("d/M/y Hmm"));
+                        String convertedTime = eventTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a"));
+                        String stringSliced = input.substring(6, indexOfTime); // after deadline
+                        stringArray.add(stringSliced);
+                        Event eventTask = new Event(stringSliced, convertedTime);
+                        taskArray.add(eventTask);
+                        String noOfTask = String.valueOf(taskArray.size());
+                        System.out.println(messageTask + eventTask.toString() + "\n"
```
Great use of comments to clarify what each line is doing!  Maybe you could also use a few blank lines to separate the code block into logical sections (such as formatting, processing and output.) The coding standard recommends that "Logical units within a block should be separated by one blank line."

manu2002g commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                            System.out.println(index + "." + taskArray.get(i));
+                        }
+                    }
+                } else if (checkType[0].equals("mark") || checkType[0].equals("unmark")) { //this is to check whether it goes through marking
```
I think this line has more than the 120 character line limit allowed by the coding standard. Maybe you could add a line break to improve readability

manu2002g commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,166 @@
+                
+                    try {
+                        int index = Integer.parseInt(checkType[1]) - 1;
+                        Task tasks = taskArray.get(index);
```
Maybe change the name of this variable to `task` since it only refers to a single entry

manu2002g reviewed (commented):
Great code quality! The variable names used are all short and descriptive and help a lot to make the code easy to understand! 😄

KristopherPTaslim committed:
Finish A-MOREOOP

kahleongq commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,129 @@
+            default:
+                ui.showDefaultMessage();
+                break;
+
```
Perhaps you could remove the indentation for the case clauses to follow the coding standards

kahleongq reviewed (commented):
Overall you adhered mostly to the coding standards, keep it up!

KristopherPTaslim committed:
Finish A-Packages

eugenechiaay commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,21 @@
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
```
Perhaps you could rename "by" to something more descriptive.

```suggestion
    public Deadline(String description, String deadlineDate) {
        super(description);
        this.deadlineDate = deadlineDate;
    }
```

eugenechiaay commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+            Parser parser = new Parser(taskList);
+            
+            while (sc.hasNextLine()) {
+                parser.execute(input);
```
perhaps you can change the name of the method since a parser is parsing an input

```suggestion
                parser.parseInput(input);
```

eugenechiaay commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+
+class DukeException extends RuntimeException {
+
+    public DukeException(String Message) {
+        super(Message);
+    }
```
You could add a more descriptive message to be stored within the DukeException class

eugenechiaay commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,21 @@
+
+    protected String at;
+
+    public Event(String description, String at) {
+        super(description);
+        this.at = at;
+    }
```
Same thing with the deadline task

```suggestion
    public Event(String description, String eventDate) {
        super(description);
        this.eventDate= eventDate;
    }
```

eugenechiaay commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,131 @@
+                    LocalDateTime eventTime = LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("d/M/y Hmm"));
+                    String convertedTime = eventTime.format(DateTimeFormatter.ofPattern("MMM d yyyy hh:mm a"));
+                    String stringSliced = input.substring(6, indexOfTime); // after deadline
+                    Event eventTask = new Event(stringSliced, convertedTime);
+                    taskList.addTask(eventTask);
+                    String noOfTask = String.valueOf(taskList.getTaskArray().size());
+                    ui.showAddedMessage(eventTask, noOfTask);
+
+                } catch (StringIndexOutOfBoundsException e) {
+                    ui.showEventError();
+                }
+                break;
+
+            case("delete"):
+
+                try {
+                    int index = Integer.parseInt(checkCase[1]) - 1;
+                    Task task = taskList.getTaskArray().get(index);
+                    taskList.deleteTask(index);
+                    String noOfTask = String.valueOf(taskList.getTaskArray().size());
+                    ui.showDeletedMessage(task, noOfTask);
+                } catch (ArrayIndexOutOfBoundsException e) {
+                    ui.showDeleteError();
+                }
+                break;
+
+            case("bye"):
+                ui.showGoodbyeMessage();
+                break;             
+
+            default:
+                ui.showDefaultMessage();
+                break;
+
+
+
+        }
+
+
+    }
```
Apart from the coding standard violations, perhaps you can break up each case instruction into separate methods for increased readability? 

For example:

```
public void delete() {
    try {
                    int index = Integer.parseInt(checkCase[1]) - 1;
                    Task task = taskList.getTaskArray().get(index);
                    taskList.deleteTask(index);
                    String noOfTask = String.valueOf(taskList.getTaskArray().size());
                    ui.showDeletedMessage(task, noOfTask);
                } catch (ArrayIndexOutOfBoundsException e) {
                    ui.showDeleteError();
                }
}
```

eugenechiaay reviewed (commented):
Hi Kris! Code is generally well done, just keep in mind that it's best to break up code into smaller blocks for increased readability and to use more descriptive naming:)

KristopherPTaslim committed:
Finish A-JUnit

KristopherPTaslim committed:
Debug A-JUnit

KristopherPTaslim committed:
Delete the wrong folder test:java:duke

KristopherPTaslim committed:
Finish A-Javadoc

KristopherPTaslim committed:
Added missing JavaDoc in the branch-A-JavaDoc

KristopherPTaslim committed:
Finish A-CodingStandard

KristopherPTaslim committed:
Finish branch-Level-9

KristopherPTaslim committed:
Merge tag 'branch-A-CodingStandard'

KristopherPTaslim committed:
Merge tag 'branch-Level-9'

KristopherPTaslim committed:
Merging all the three branches(JavaDoc, Coding Standard and Level 9)

KristopherPTaslim committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d' into branch-A-Gradle

KristopherPTaslim committed:
Finish running branch-A-Gradle

KristopherPTaslim committed:
Finish Level-10

KristopherPTaslim committed:
Fix Minor Bug

KristopherPTaslim committed:
Edit execute() method in Parser class to follow coding standard.

The method execute() does not follow the coding standard due to its
length. There are common logic in the method which cause
code duplication.

Extracting this common logic allows the method to be significantly
shorter and the principle of SLAP to be followed. For example,
the method parseByCondition() and parseString() shows the
SLAP principle.

KristopherPTaslim committed:
Add assertions in some various points in code

Assertions are required in order to check whether the code
is able to work.

Assertions allow assertion error to be thrown if some conditions
are not held properly. For example, implementing assertion task
cannot be null in the method showAddedMessage(Task task, String
noOfTask) will throw assertion error if the input of task is null.

KristopherPTaslim committed:
Merge pull request #2 from KristopherPTaslim/branch-A-Assertions

Add assertions in some various points in code

KristopherPTaslim committed:
Merge branch 'master' into branch-A-CodeQuality

KristopherPTaslim committed:
Merge pull request #3 from KristopherPTaslim/branch-A-CodeQuality

Edit execute() method in Parser class to follow coding standard.

KristopherPTaslim committed:
Edit minor feature in Parser class

KristopherPTaslim committed:
Finish branch-BCD-Extension

KristopherPTaslim committed:
Add Ui.png under the docs folder

KristopherPTaslim committed:
Update ReadME.md

KristopherPTaslim committed:
Update ReadME.md

KristopherPTaslim committed:
Edit some parts in build.gradle
